cpp/include/cudf/functions.h

```diff
@@ -820,114 +2534,345 @@ gdf_error gdf_max(gdf_column *col, void *dev_result, gdf_size_type dev_result_si
+  *
+* @Returns GDF_SUCCESS upon successful compute, otherwise returns appropriate error code
+ */
+/* ----------------------------------------------------------------------------*/
+gdf_error gdf_group_by_avg(int ncols,
+                           gdf_column** cols,
+                           gdf_column* col_agg,
+                           gdf_column* out_col_indices,
+                           gdf_column** out_col_values,
+                           gdf_column* out_col_agg,
+                           gdf_context* ctxt);
+
+/**
+ * @brief  Performs SQL like GROUP BY with COUNT aggregation
+ *
+ * @param[in] # columns
+ * @param[in] input cols
+ * @param[in] column to aggregate on
+ * @param[out] if not null return indices of re-ordered rows
+ * @param[out] if not null return the grouped-by columns (multi-gather based on indices, which are needed anyway)
+ * @param[out] aggregation result
+ * @param[in] struct with additional info: bool is_sorted, flag_sort_or_hash, bool flag_count_distinct
+  *
+* @Returns GDF_SUCCESS upon successful compute, otherwise returns appropriate error code
+ */
+/* ----------------------------------------------------------------------------*/
+gdf_error gdf_group_by_count(int ncols,
+                             gdf_column** cols,
+                             gdf_column* col_agg,
+                             gdf_column* out_col_indices,
+                             gdf_column** out_col_values,      
+                             gdf_column* out_col_agg,
+                             gdf_context* ctxt);
+
+/**
+ * @brief  Calculates exact quantiles
+ *
+ * @param[in] input column
+ * @param[in] precision: type of quantile method calculation
+ * @param[in] requested quantile in [0,1]
+ * @param[out] result; for <exact> should probably be double*; it's void* because: (1) for uniformity of interface with <approx>; (2) for possible types bigger than double, in the future;
+ * @param[in] struct with additional info
+  *
+* @Returns GDF_SUCCESS upon successful compute, otherwise returns appropriate error code
+ */
+/* ----------------------------------------------------------------------------*/
+gdf_error gdf_quantile_exact(gdf_column* col_in,
+                            gdf_quantile_method prec,
+                            double q,
+                            void* t_erased_res,                            
+                            gdf_context* ctxt);
+
+/**
+ * @brief  Calculates approximate quantiles
+ *
+ * @param[in] input column
+ * @param[in] requested quantile in [0,1]
+ * @param[out] result; type-erased result of same type as column;
+ * @param[in] struct with additional info
+  *
+* @Returns GDF_SUCCESS upon successful compute, otherwise returns appropriate error code
+ */
+/* ----------------------------------------------------------------------------*/
+gdf_error gdf_quantile_aprrox(gdf_column* col_in,
```

Comment:
Is this a typo? The function name should be `gdf_quantile_approx` (two 'p's, one 'r', as in the word approximation).